packages/Caching/Detector/ChangedFilesDetector.php

```diff
@@ -35,11 +35,15 @@ public function addFileWithDependencies(string $filePath, array $dependentFiles)
 
     public function hasFileChanged(string $filePath): bool
     {
-        $currentFileHash = $this->hashFile($filePath);
         $fileInfoCacheKey = $this->getFilePathCacheKey($filePath);
-
         $cachedValue = $this->cache->load($fileInfoCacheKey, CacheKey::FILE_HASH_KEY);
-        return $currentFileHash !== $cachedValue;
+
+        if ($cachedValue !== null) {
+            $currentFileHash = $this->hashFile($filePath);
+            return $currentFileHash !== $cachedValue;
+        }
+
+        return true;
```

Comment:
could you give some comment here? just by direct read, it little bit confusing

Reply:
The logic of the return value didn‘t change with this PR. I added a comment